Src/Common/FSecure/AsanaApi/AsanaApi.cpp

```diff
@@ -0,0 +1,196 @@
+	}
+	// Building the multipart body (prefix + attachment + suffix)
+	std::vector<uint8_t> body;
+	std::string bodyPrefix = OBF("\r\n");
+	bodyPrefix += OBF("--") + boundary + OBF("\r\n");
+	bodyPrefix += OBF("Content-Disposition: form-data; name=\"file\"; filename=\"") + attachmentFileName + OBF("\"") + OBF("\r\n");
+	bodyPrefix += OBF("Content-Type: ") + attachmentMimeType + OBF("\r\n\r\n");
+	body.insert(body.begin(), bodyPrefix.begin(), bodyPrefix.end()); // Insert the prefix
+	body.insert(body.end(), attachmentBody.begin(), attachmentBody.end()); // Insert the attachment content
+	std::string bodySuffix = OBF("\r\n");
+	bodySuffix += OBF("--") + boundary + OBF("--") + OBF("\r\n");
+	body.insert(body.end(), bodySuffix.begin(), bodySuffix.end()); // Insert the suffix
+	// Send HTTP request
+	std::string contentType = OBF("multipart/form-data; boundary=") + boundary;
+	json response = json::parse(SendHttpRequest(url, { contentType.begin(), contentType.end() }, body, Method::POST));
+	return response[OBF("data")][OBF("gid")];
+}
+
+void FSecure::AsanaApi::RenameTask(std::string const& taskId, std::string const& newTaskName) {
+	std::string url = OBF("https://app.asana.com/api/1.0/tasks/") + taskId;
+	json j;
+	j[OBF("data")][OBF("name")] = newTaskName;
+	SendJsonRequest(url, j, Method::PUT);
+}
+
+std::vector<uint8_t> FSecure::AsanaApi::GetAttachmentById(std::string const& attachmentId) {
+	// First we need to get the download url (Asana stores its attachments in S3)
+	std::string url = OBF("https://app.asana.com/api/1.0/attachments/") + attachmentId;
+	json response = SendJsonRequest(url, NULL, Method::GET);
+	std::string downloadUrl = response[OBF("data")][OBF("download_url")];
+	// Then download the content of the attachment
+	std::string::size_type i = downloadUrl.find(OBF("#_=_")); // Got to remove the '#_=_' of the download string (cpprestsdk can't handle this)
+	if (i != std::string::npos)
+		downloadUrl.erase(i, 4);
+	ByteVector content = SendHttpRequest(downloadUrl, ContentType::Text, {}, Method::GET, false); // Content type will be ignored, since data is empty
+	return std::vector<uint8_t>(std::make_move_iterator(content.begin()), std::make_move_iterator(content.end()));
+}
+
+FSecure::ByteVector FSecure::AsanaApi::SendHttpRequest(std::string const& host, FSecure::WinHttp::ContentType contentType, std::vector<uint8_t> const& data, FSecure::WinHttp::Method method, bool setAuthorizationHeader) {
+	return SendHttpRequest(host, GetContentType(contentType), data, method, setAuthorizationHeader);
+}
+
+FSecure::ByteVector FSecure::AsanaApi::SendHttpRequest(std::string const& host, std::wstring contentType, std::vector<uint8_t> const& data, FSecure::WinHttp::Method method, bool setAuthorizationHeader) {
+	while (true) {
+		HttpClient webClient(ToWideString(host), m_ProxyConfig);
+		HttpRequest request;
+		request.m_Method = method;
+
+		if (!data.empty()) {
+			request.SetData(contentType, data);
+		}
```

Comment:
[Style] brackets - This `if` has brackets, the next one doesn't, and the last `if - else` uses both. Use one style for single line statements.

Reply:
Resolved in 59b55f11f75d62233d1781cc5e2a28dbe0cc7967.